src/vtkF3DAlembicImporter.cxx

```diff
@@ -0,0 +1,351 @@
+
+		//set this if we should continue traversing
+		Alembic::Abc::IObject nextParentObject;
+
+		if ( Alembic::AbcGeom::IXform::matches( ohead ) )
+		{
+			std::cout << "iterate_iobject match IXform" << std::endl;
+			Alembic::AbcGeom::IXform xform( parent, ohead.getName() );
+
+			nextParentObject = xform;
+
+		}
+		else if ( Alembic::AbcGeom::ISubD::matches( ohead ) )
+		{
+			std::cout << "iterate_iobject match ISubD" << std::endl;
+			Alembic::AbcGeom::ISubD subd( parent, ohead.getName() );
+			nextParentObject = subd;
+
+		}
+		else if ( Alembic::AbcGeom::IPolyMesh::matches( ohead ) )
+		{
+			std::cout << "iterate_iobject match IPolyMesh" << std::endl;
+			Alembic::AbcGeom::IPolyMesh polymesh( parent, ohead.getName() );
+			ProcessIPolyMesh(renderer, polymesh);
+			nextParentObject = polymesh;
+
+		}
+		else if ( Alembic::AbcGeom::INuPatch::matches( ohead ) )
+		{
+			std::cout << "iterate_iobject match INuPatch" << std::endl;
+
+		}
+		else if ( Alembic::AbcGeom::IPoints::matches( ohead ) )
+		{
+			std::cout << "iterate_iobject match IPoints" << std::endl;
+			Alembic::AbcGeom::IPoints points( parent, ohead.getName() );
+			// ProcessIPoints(points);
+			nextParentObject = points;
+		}
+		else if ( Alembic::AbcGeom::ICurves::matches( ohead ) )
+		{
+			std::cout << "iterate_iobject match ICurves" << std::endl;
+
+		}
+		else if ( Alembic::AbcGeom::IFaceSet::matches( ohead ) )
+		{
+			std::cout << "iterate_iobject match IFaceSet" << std::endl;
+			std::cerr << "DOH !" << std::endl;
+		}
+
+		// Recursion
+		if ( nextParentObject.valid() )
+		{
+			for ( size_t i = 0; i < nextParentObject.getNumChildren() ; ++i )
+			{
+				IterateIObject( renderer, nextParentObject, nextParentObject.getChildHeader( i ));
+			}
+		}
+
+	}
+	void ImportRoot(vtkRenderer* renderer)
+	{
+		Alembic::Abc::IObject top = _archive.getTop();
+		size_t top_num_children = top.getNumChildren();
+
+		//   vtkDebugMacro("Done with " << this->GetClassName() << "::" << __FUNCTION__);
+		std::cout <<  top_num_children << std::endl;
+
+	    for ( size_t i = 0; i < top.getNumChildren(); ++i )
+	    {
+	    	IterateIObject(renderer, top, top.getChildHeader(i));
+	    }
+
+	}
+	void ImportCameras(vtkRenderer* renderer)
+	{
+
+	}
+	void ReadScene(const std::string& filePath)
+	{
+
+		Alembic::AbcCoreFactory::IFactory factory;
+		Alembic::AbcCoreFactory::IFactory::CoreType core_type;
+
+		_archive = factory.getArchive(filePath, core_type);
+
+	}
+	std::string Description;
+	std::vector<vtkSmartPointer<vtkPolyData> > Meshes;
+	std::vector<vtkSmartPointer<vtkProperty> > Properties;
+	std::vector<vtkSmartPointer<vtkTexture> > EmbeddedTextures;
+	vtkIdType ActiveAnimation = 0;
+	std::vector<std::pair<std::string, vtkSmartPointer<vtkLight> > > Lights;
+	std::vector<std::pair<std::string, vtkSmartPointer<vtkCamera> > > Cameras;
+	vtkIdType ActiveCameraIndex = -1;
+	std::unordered_map<std::string, vtkSmartPointer<vtkActorCollection> > NodeActors;
```

Comment:
a lot of these fields are copied from Assimp and are probably useless for you.

Reply:
I have remove all the unused member variables.